tests/phpunit/tests/fonts/font-face/wp-font-face-tests-dataset.php

```diff
@@ -206,7 +206,7 @@ public function get_expected_fonts_for_fonts_block_theme( $key = '' ) {
 			$uri  = get_stylesheet_directory_uri() . '/assets/fonts/';
 			$data = array(
 				'fonts'            => array(
-					'DM Sans'          => array(
```

Comment:
Instead of changing this dataset, I suggest adding a new dataset to cover the use case specifics for this PR. The specifics being: no font-family key requirement and a font in different origins.

Why keep this dataset?
Test coverage for BC and use cases where `wp_print_font_faces()` is directly invoked in a plugin or theme (such as classic sites) and passed an array of fonts to print. 

Notice, `wp_print_font_faces()` is documented to require a font-family key. This docblock will need to change in this PR too to make the key optional.

Reply:
Adding a new dataset may also require adding a new set of tests.

Reply:
> Instead of changing this dataset
This particular change you highlighted here should remain as in this PR because the only data changed is the expected output and not the function's input.

> I suggest adding a new dataset to cover the use case specifics for this PR
Good idea;  I'll add a test case covering that the function continues working as expected for retro-compatibility, and it works as expected when more than one origin is present in theme.json data.


> Notice, wp_print_font_faces() is documented to require a font-family key. This docblock will need to change in this PR too to make the key optional.

Thank you for pointing that out. I'll update that,

Reply:
Recap of the latest commit according to your feedback:
- Add a test case for the bug this PR is fixing (font families defined in more than one theme.json origin): https://github.com/WordPress/wordpress-develop/pull/6161/commits/66f6bc80ae3c996053fbcddb1b33630c54dacf0a
- Add a test case for using an indexed array as parameter for wp_print_font_faces: https://github.com/WordPress/wordpress-develop/pull/6161/commits/c3765fe8a44b80f72bb51b9c5ccd0f1b2763aaab
- Update wp_print_font_faces docblock https://github.com/WordPress/wordpress-develop/pull/6161/commits/a4b0ff465df60b84229ea4e473620682872b5eab

I think it's ready to review